runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -42,161 +53,211 @@ public final class LogReplicationUtils {
+                                                .equals(LogReplication.ReplicationModel.LOGICAL_GROUPS) &&
+                                                Objects.equals(entry.getKey().getSubscriber().getClientName(),
+                                                        listener.getClientName()));
+
+                        CorfuStoreEntry<LogReplicationSession, ReplicationStatus, Message> entry = null;
+
+                        // It is possible that there is no entry for the Logical Group Model in the status table in
+                        // the following scenarios:
+                        // 1. Request to subscribe is received before the Log Replication JVM or pod starts and
+                        // initializes the table
+                        // 2. Topology change which removes this cluster from the status table.
+                        // We cannot differentiate between the two cases here so continue with the right
+                        // behavior for the 1st case, i.e., subscribe and wait for the initial Snapshot Sync to get
+                        // triggered.  If we are here because of the second case, the listener will not get any updates
+                        // from LR and subscription will be a no-op.
+                        if (entries.isEmpty()) {
+                            log.warn("No record for client {} and Logical Group Model found in the Status Table.  " +
+                                    "Subscription could have been attempted before LR startup.  Subscribe the listener and" +
+                                    "wait for initial Snapshot Sync", listener.getClientName());
+                        } else {
+                            // For a given replication model and client, any Sink node will have a single session.
+                            Preconditions.checkState(entries.size() == 1);
+                            entry = entries.get(0);
+                        }
+
+                        if (entry == null || entry.getPayload().getSinkStatus().getDataConsistent()) {
+                            // No snapshot sync is in progress
+                            log.info("No Snapshot Sync is in progress.  Request the client to perform a full sync on its " +
+                                    "tables.");
+                            Optional<Timer.Sample> clientFullSyncTimer = MicroMeterUtils.startTimer();
+                            long clientFullSyncStartTime = System.currentTimeMillis();
+                            listener.performFullSyncAndMerge(txnContext);
+                            long clientFullSyncEndTime = System.currentTimeMillis();
+                            MicroMeterUtils.time(clientFullSyncTimer, "logreplication.client.fullsync.duration");
+                            log.info("Client Full Sync and Merge took {} ms",
+                                    clientFullSyncEndTime - clientFullSyncStartTime);
+                            txnContext.commit();
```

Comment:
nit: can we also log the return value from the commit() tells us how far the subscription and re-write are?